clang/lib/CodeGen/HLSLBufferLayoutBuilder.cpp:92: Possible && || mixup ?

A recent build of clang by gcc says:

clang/lib/CodeGen/HLSLBufferLayoutBuilder.cpp:92:58: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]

Source code is

      assert(!Packoffsets || Index < Packoffsets->size() &&
                                 "number of elements in layout struct does not "
                                 "match number of packoffset annotations");

Suggest add ( and ) to make intention clear.

--- a/clang/lib/CodeGen/HLSLBufferLayoutBuilder.cpp
+++ b/clang/lib/CodeGen/HLSLBufferLayoutBuilder.cpp
@@ -89,9 +89,9 @@ llvm::TargetExtType *HLSLBufferLayoutBuilder::createLayoutType(
     RecordTypes.pop_back();
 
     for (const auto *FD : RT->getDecl()->fields()) {
-      assert(!Packoffsets || Index < Packoffsets->size() &&
-                                 "number of elements in layout struct does not "
-                                 "match number of packoffset annotations");
+      assert((!Packoffsets || Index < Packoffsets->size()) &&
+             "number of elements in layout struct does not "
+             "match number of packoffset annotations");
 
       if (!layoutField(FD, EndOffset, Layout, LayoutElements,
                        Packoffsets ? (*Packoffsets)[Index] : -1))

